src/main/kotlin/net/shadowfacts/forgelin/Forgelin.kt

```diff
@@ -12,4 +16,10 @@ object Forgelin {
 	const val NAME = "Forgelin"
 	const val VERSION = "@VERSION@"
 
-}
\ No newline at end of file
+	@EventHandler
+	fun onPreInit(event: FMLPreInitializationEvent) {
```

Comment:
unless I'm reading it wrong, forge seems to do it during construct... could you do the same?

Reply:
Not sure if it is possible to hook into construction -- when I checked, it didn't look like it. Though I suppose it could be possible by running within the constructor?

Reply:
You need the ASM data, which I don't believe is available until preinit to mods.

Reply:
you should have it:
```java
public class FMLConstructionEvent extends FMLStateEvent
{
    private ModClassLoader modClassLoader;
    private ASMDataTable asmData;
    private ListMultimap<String,String> reverseDependencies;
```